M3-4236 EntityDetail component

johnwcallahan opened the pull request:
## Description

This PR adds a new component which I'm calling "EntityDetail". It's similar to the LKE Summary panel. In CMR, it's found on Linode Details, Linodes Landing, and the Dashboard (and will conceivably be used with other entity types as well).

![Screen Shot 2020-06-24 at 2 03 42 PM](https://user-images.githubusercontent.com/16911484/85611356-e0377f00-b625-11ea-973c-5e2f28d8b0fe.png)

The new component is a "shell" component, meaning the consumer provides all the pieces; this component is only responsible for positioning them correctly.

I also created a `<LinodeEntityDetail />` component, which is a wrapper around EntityDetail, providing the pieces we need for Linodes. 

Also:
- Removed some of the Grid's in DetailHeader that were causing me some trouble with negative margins.
- Removed the `<Paper />`from DetailHeader (in this new component, everything sits on top of one Paper).
- Added a new utility CSS class: `flexCenter`.

Still todo:
- "Tags" isn't implemented yet... that'll come later.
- Would like to get a few unit tests together for this.
- Not styled for mobile yet.

## Note to Reviewers

Currently this component is in Storybook only: http://localhost:6006/?path=/story/entitydetail--linode

Jskobos commented on packages/manager/src/components/EntityDetail/EntityDetail.tsx:
```diff
@@ -0,0 +1,58 @@
+
+const useStyles = makeStyles(() => ({
+  header: {
+    padding: `0px 20px 0px 0px !important`
```
I'll defer to @WilkinsKa1 on this, imo it seems a little iffy to have `!important` so high up in a component tree.

WilkinsKa1 replied:
If I remove the `important`, the padding persists for me. We could also use paddingRight for this instance.

Jskobos commented on packages/manager/src/features/linodes/LinodeEntityDetail.tsx:
```diff
@@ -0,0 +1,421 @@
+      <Grid item className={classes.ipContainer}>
+        <Typography className={classes.ipLabel}>IP Addresses</Typography>
+        <List className={classes.ipList}>
+          {ipv4.map(thisIP => {
```
I know many IPs is a rare case, but as a safety check maybe slice this after a certain point.

DevDW replied:
Maybe top it out at four? I do vaguely recall a mention from one of the CMR meetings of possibly having a scrollable element they are listed in when the list is long, but I might be mistaken.

DevDW replied:
Maybe top it out at four? I do vaguely recall a mention from one of the CMR meetings of possibly having a scrollable element they are listed in when the list is long, but I might be mistaken.

Jskobos commented on packages/manager/src/features/linodes/LinodeEntityDetail.tsx:
```diff
@@ -0,0 +1,421 @@
+}));
+
+export const Body: React.FC<BodyProps> = React.memo(props => {
+  const {
```
flat + memo'd, 💯

WilkinsKa1 commented on packages/manager/src/components/EntityHeader/HeaderBreadCrumb.tsx:
```diff
@@ -17,6 +16,10 @@ const useStyles = makeStyles((theme: Theme) => ({
     position: 'relative',
     backgroundColor: theme.bg.lightBlue,
     marginRight: theme.spacing(2),
+    paddingTop: 5,
```
If we're not overriding anything here, we could use shorthand: padding: `5px ${theme.spacing()}px`

WilkinsKa1 commented on packages/manager/src/features/linodes/LinodeEntityDetail.tsx:
```diff
@@ -0,0 +1,421 @@
+    marginRight: 50
+  },
+  iconSharedOuter: {
+    textAlign: 'center',
```
The icons should be vertically aligned, I think if you add justify-content: center here you should be g2g 👍

johnwcallahan commented:
Thanks @Jskobos, @WilkinsKa1, and @DevDW for the feedback.

- Added dynamic distro icons
- Added "View Details", "Power Off", and "Reboot" buttons
- Display max of 4 IPs with a "... plus X more" link to the networking tab.

Huge thanks to @WilkinsKa1 who helped me troubleshoot some of the Grid/Spacing issues I faced after rebasing.

johnwcallahan commented:
@Jskobos I'll address the dark mode feedback when it's implemented in the app for real.

WilkinsKa1 commented on packages/manager/src/features/linodes/LinodeEntityDetail.tsx:
```diff
@@ -0,0 +1,514 @@
+      </Grid>
+
+      <Grid item className={classes.accessTableContainer}>
+        <Table className={classes.accessTable}>
```
We can come back to this, but we should have a table header for the cell headings. We can change the orientation of the th as needed as well.

tiffwong commented:
Looks good! 
- A few things, the Linode name is in black in the designs (which makes more sense since blue makes it look like a link)
- The distance between the "Reboot" and "More Actions" button is too far. The design shows that it is 30px apart but the "More Actions" icon in the code has a 10px padding 
  <img width="266" alt="Screen Shot 2020-06-25 at 2 49 41 PM" src="https://user-images.githubusercontent.com/7692354/85780902-1e4ea480-b6f3-11ea-9636-fa80ddcf348a.png">
- The font weight for the Linode ID also needs to be increased to 500

Jskobos commented on packages/manager/src/features/linodes/LinodeEntityDetail.tsx:
```diff
@@ -75,16 +95,67 @@ interface HeaderProps {
       actions={
-        <ActionMenu ariaLabel="linode-detail" createActions={() => []} />
+        <>
+          <IconTextLink
```
This might be better if it's conditional, since it'll appear in the "grid" view but not on the actual detail view
